packages/arb-bridge-eth/src/bridge.ts

```diff
@@ -0,0 +1,214 @@
+  for (let i = 0; i < completeData.length; i++) {
+    if (i < zeroBytesToFillPrefix) {
+      completeData[i] = 0;
+    } else {
+      completeData[i] = event.params.data[i - zeroBytesToFillPrefix];
+    }
+  }
+
+  // decode it and save to EthDeposit entity
+  const decodedData = ethereum.decode("(address,uint256)", completeData);
+  if (decodedData) {
+    const decodedTuple = decodedData.toTuple();
+    entity.destAddr = decodedTuple[0].toAddress();
+    entity.value = decodedTuple[1].toBigInt();
+  }
+  entity.isClassic = false;
+  entity.timestamp = event.block.number;
+  entity.transactionHash = event.transaction.hash;
+  entity.blockCreatedAt = event.block.number;
+  entity.save();
+
+  // delete the old raw message
+  store.remove("RawMessage", id);
+}
+
+function handleNitroRetryable(event: InboxMessageDeliveredEvent, rawMessage: RawMessage): void {
+  const id = bigIntToId(event.params.messageNum);
+  const retryable = RetryableTx.parseRetryable(event.params.data);
+  if (retryable) {
+    let entity = new Retryable(id);
+    // get sender from preceding MessageDelivered event
+    entity.senderAliased = rawMessage.sender;
+    entity.value = event.transaction.value;
+    entity.isEthDeposit = retryable.dataLength == BigInt.zero();
+    entity.retryableTicketID = getL2RetryableTicketId(event.params.messageNum);
+    entity.destAddr = retryable.destAddress;
+    entity.l2Calldata = retryable.data;
+    entity.timestamp = event.block.timestamp;
+    entity.transactionHash = event.transaction.hash;
+    entity.blockCreatedAt = event.block.number;
+    entity.save();
+    // we delete the old raw message since now we saved the retryable
+    store.remove("RawMessage", id);
+  } else {
+    log.error("Not able to parse tx with id {}", [id.toString()]);
+  }
+}
+
+function handleClassicRetryable(
+  event: InboxMessageDeliveredEvent,
+  rawMessage: ClassicRawMessage
+): void {
+  const id = bigIntToId(event.params.messageNum);
+  const retryable = RetryableTx.parseRetryable(event.params.data);
+  if (retryable) {
+    // if there is no calldata, this is Eth deposit
+    if (retryable.dataLength == BigInt.zero()) {
+      // we track deposits with EthDeposit entities
+      let depositId =
+        event.transaction.hash.toHexString() + "-" + event.transaction.index.toString();
+      let deposit = new EthDeposit(depositId);
+      deposit.senderAliased = rawMessage.sender;
+      deposit.msgData = event.params.data;
+      deposit.destAddr = retryable.destAddress;
+      deposit.value = event.transaction.value;
+      deposit.isClassic = true;
+      deposit.timestamp = event.block.number;
+      deposit.transactionHash = event.transaction.hash;
+      deposit.blockCreatedAt = event.block.number;
+      deposit.save();
+    }
+
+    let entity = new ClassicRetryable(id);
+    // get sender from preceding MessageDelivered event
+    entity.senderAliased = rawMessage.sender;
```

Comment:
Classic event should contain the sender address without alias here; the aliasing is done in arbos (unlike nitro the event contain the aliased address)